### ISSUE TITLE
Generic Wayland icon appears for all apps in the suite ONLY in the title bar in KDE Plasma

### ISSUE THREAD

**Issue by n3thshan** [flathub/org.libreoffice.LibreOffice#312]
Proper icons are used in the task bar. Currently the desktop file name workaround in Plasma Window Rules does not work either. 

![image](https://github.com/user-attachments/assets/48c65b3c-73cc-46ca-acef-c5db3b0b95d0)

**Comment by n3thshan** [comment]
hello? why isnt anyone active here? this apparently can be simply fixed by setting `"copy-icon": true` in the manifest as per [here](https://github.com/PapirusDevelopmentTeam/papirus-icon-theme/issues/1300#issuecomment-394278316). Please fix this. Thank you!

**Comment by n3thshan** [comment]
the same issue exists in [org.tenacityaudio.Tenacity](https://github.com/flathub/org.tenacityaudio.Tenacity). the collaborator there said he'd get back when identified the root cause. i shall open a pr then.

**Comment by n3thshan** [comment]
> Did you test if https://github.com/flathub/org.tenacityaudio.Tenacity/pull/35 fixed this issue? For me it didn't change anything. This is wayland specific issue and spotify issue you mention wasn't related to wayland.

yes, the copy-icon fix did not work for me either.

> You may read more about the root cause https://github.com/flathub/com.discordapp.Discord/pull/354

i tried the KDE window rules workaround but that didnt fix it either for me. Cannot this fix be implemented for libreoffice as well?

**Comment by Erick555** [comment]
I don't think it can. It involves renaming binary to match the icon name but libreoffice uses same binary for launching multiple apps (calc, writer, etc.).

The workaround is forcing xwayland by running `flatpak override --user --nosocket=wayland org.libreoffice.LibreOffice`

**Comment by n3thshan** [comment]
@Erick555 thanks for the workaround but what is the catch/s using xwayland?

**Comment by Erick555** [comment]
The catch is the app will use legacy xorg server to display its windows instead of modern wayland server.

**Comment by n3thshan** [comment]
okay after some testing i found that KDE's window rule fix works when launching writer, calc, etc... through the start center. However, when directly opening writer, calc, etc... it does not apply.

**Comment by sibelman** [comment]
Just a gratitude post:  great exploration of this curious issue!  The workaround @Erick555 posted is splendid, and opens the door for me to learn more about fine-tuning flatpaks.

**Comment by sibelman** [comment]
The solution Erick555 posted may work for some, but[ this solution](https://starkle.dev/fix-libreoffice-flatpak-wayland-icons-on-kde/) was the only one I found that thoroughly addressed the generic Wayland icon issue for LibreOffice.  Erick's caused launch failure in my particular (Fedora Kinoite 44) environment.  Erick's worked well in the past though, and may still suffice in many/most instances.
